File: backend/Raporlar/OdemeDurumu/Dto/OdemeDurumuRaporDto.cs
namespace STYS.Raporlar.OdemeDurumu.Dto;

public class OdemeDurumuRaporDto
{
    public int TesisId { get; set; }
    public string? TesisAdi { get; set; }
    public DateTime Baslangic { get; set; }
    public DateTime Bitis { get; set; }
    public string OdemeDurumu { get; set; } = "borclu";

    public OdemeDurumuOzetDto Ozet { get; set; } = new();
    public List<OdemeDurumuRezervasyonDto> Rezervasyonlar { get; set; } = [];
}

public class OdemeDurumuOzetDto
{
    public int ToplamRezervasyonSayisi { get; set; }
    public int BorcluRezervasyonSayisi { get; set; }
    public int OdemesiOlmayanRezervasyonSayisi { get; set; }
    public int KismiOdendiRezervasyonSayisi { get; set; }
    public int TamamenOdendiRezervasyonSayisi { get; set; }
    public int CikisYapmisBorcluRezervasyonSayisi { get; set; }
    public decimal ToplamUcret { get; set; }
    public decimal ToplamOdenenTutar { get; set; }
    public decimal ToplamKalanTutar { get; set; }
    public string ParaBirimi { get; set; } = "TRY";
}

public class OdemeDurumuRezervasyonDto
{
    public int RezervasyonId { get; set; }
    public string ReferansNo { get; set; } = string.Empty;
    public string MisafirAdiSoyadi { get; set; } = string.Empty;

    // TODO: Kurum/unite bilgisi mevcut rezervasyon veri modelinde tutulmuyor; ileride eklenirse doldurulacak.
    public string? KurumUnite { get; set; }

    public DateTime GirisTarihi { get; set; }
    public DateTime CikisTarihi { get; set; }
    public string RezervasyonDurumu { get; set; } = string.Empty;
    public string RezervasyonDurumuLabel { get; set; } = string.Empty;
    public List<string> OdaNolari { get; set; } = [];
    public int KisiSayisi { get; set; }

    public decimal ToplamUcret { get; set; }
    public decimal OdenenTutar { get; set; }
    public decimal KalanTutar { get; set; }
    public string ParaBirimi { get; set; } = "TRY";

    public string OdemeDurumu { get; set; } = string.Empty;
    public string OdemeDurumuLabel { get; set; } = string.Empty;
    public DateTime? SonOdemeTarihi { get; set; }
    public int OdemeSayisi { get; set; }

    public bool BorcluMu { get; set; }
    public bool CikisYapmisMi { get; set; }
    public bool CikisYapmisBorcluMu { get; set; }
}

File: frontend/src/app/pages/raporlar/odeme-durumu/odeme-durumu-rapor.dto.ts
export interface OdemeDurumuOzetDto {
    toplamRezervasyonSayisi: number;
    borcluRezervasyonSayisi: number;
    odemesiOlmayanRezervasyonSayisi: number;
    kismiOdendiRezervasyonSayisi: number;
    tamamenOdendiRezervasyonSayisi: number;
    cikisYapmisBorcluRezervasyonSayisi: number;
    toplamUcret: number;
    toplamOdenenTutar: number;
    toplamKalanTutar: number;
    paraBirimi: string;
}

export interface OdemeDurumuRezervasyonDto {
    rezervasyonId: number;
    referansNo: string;
    misafirAdiSoyadi: string;
    kurumUnite: string | null;
    girisTarihi: string;
    cikisTarihi: string;
    rezervasyonDurumu: string;
    rezervasyonDurumuLabel: string;
    odaNolari: string[];
    kisiSayisi: number;
    toplamUcret: number;
    odenenTutar: number;
    kalanTutar: number;
    paraBirimi: string;
    odemeDurumu: string;
    odemeDurumuLabel: string;
    sonOdemeTarihi: string | null;
    odemeSayisi: number;
    borcluMu: boolean;
    cikisYapmisMi: boolean;
    cikisYapmisBorcluMu: boolean;
}

export interface OdemeDurumuRaporDto {
    tesisId: number;
    tesisAdi: string | null;
    baslangic: string;
    bitis: string;
    odemeDurumu: string;
    ozet: OdemeDurumuOzetDto;
    rezervasyonlar: OdemeDurumuRezervasyonDto[];
}
